### DISCUSSION TITLE
support for multiple users

### DISCUSSION THREAD

**Discussion by Abbode** [mitmproxy/mitmproxy#6563] [Q&A]
Hello,
Does it have support for multiple users to log in on the same port?
And does it save the users' username & password in a configuration file?

Thanks.

**Comment by rosydawn6** [comment]
I am not exactly sure what you mean by multiple users logging into the same port.  Do you mean using proxy authentication via username/password?  If so see https://docs.mitmproxy.org/stable/concepts-options/ under proxyauth.  I have done this before.  Nothing is saved by default to a config file but you can pass in for example an Apache htpasswd file.  In an appropriate event hook you could obtain the currently validated user via: 

`(user,password) = flow.metadata.get("proxyauth",(None,None))`

**Reply by Abbode** [reply]
yes, I meant username/password authentication.  
I'm looking for something encrypted, to avoid censorship, does mitproxy provide some encryption?  
Thanks.
